cmake/README.md

```diff
@@ -0,0 +1,55 @@
+## CMake for Mbed OS
+
+Until further notice, CMake for Mbed OS is still in development. 
```

Comment:
Could you add a blurb about:
* what this is
* why we're doing it
* what's the plan (what's there, what's being worked on)

Reply:
We will add more details

Reply:
@donatieng 
Would a link to [this](https://os.mbed.com/blog/entry/Introducing-the-new-Mbed-Tools/) be satisfactory?

Reply:
A link would be good, but additionally I think we need a concise paragraph, focused on the CMake work.

Reply:
> > We do not want to generate them but rather have components statically defined (similar to why not using GLOB in your CMake).
> 
> @0xc0170 yes of course, what I meant was have a tool for the developer to make sure we don't forget anything when refactoring for example. List all the files statically in the right order so you can just copy paste if not include. Not something that would run dynamically each time. You'd run it from your CLI before making a PR if you will.
> 
> But I guess if you miss something, CMake will tell you.

@ladislas 
Thanks for your continued interest in Mbed OS. This is still very much a work in progress and we welcome all contributions that will improve it.

Reply:
Please see: https://github.com/ARMmbed/mbed-os/pull/13589